pkgs/development/tools/misc/stlink/default.nix

```
@@ -1,17 +1,19 @@
-{ stdenv, fetchurl, cmake, libusb1 }:
+{ stdenv, fetchFromGitHub, cmake, libusb1 }:
 
 # IMPORTANT: You need permissions to access the stlink usb devices. 
 # Add services.udev.pkgs = [ pkgs.stlink ] to your configuration.nix
 
 let
-  version = "1.3.0";
+  version = "1.5.1";
 in
 stdenv.mkDerivation {
   name = "stlink-${version}";
 
-  src = fetchurl {
-    url = "https://github.com/texane/stlink/archive/${version}.tar.gz";
-    sha256 = "3e8cba21744d2c38a0557f6835a05189e1b98202931bb0183d22efc462c893dd";
+  src = fetchFromGitHub {
+    owner = "texane";
+    repo = "stlink";
```

if you decide to use my other pname recommendation
```suggestion
    repo = pname;
```

---

I think that's overboard. Github and nixpkgs are different namespaces, there is no need to bind them other than explicitly.

---

i think it's more that the repo name and package name are likely to never change, so having them defined as the same string is okay. The `fetchFromGitHub` function just wants a string for repo, so it works either way.

---

I think it will be more of a mess for someone reading this code later: `repo = pname` implies abstraction, but there is none.

If it is common practice in `nixpkgs` nowadays, then fine, else I'm against it.

---

I agree with @regellosigkeitsaxiom